anony.gemspec

```diff
@@ -24,15 +24,6 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = ">= 2.6"
 
-  spec.add_development_dependency "bundler", "~> 2"
-  spec.add_development_dependency "gc_ruboconfig", "~> 3.6.0"
-  spec.add_development_dependency "rspec", "~> 3.9"
-  spec.add_development_dependency "rspec-github", "~> 2.4.0"
-  spec.add_development_dependency "yard", "~> 0.9.20"
-
-  # For integration testing
-  spec.add_development_dependency "sqlite3", "~> 1.6.1"
```

Comment:
I believe we'd want to keep dependencies in the `gemspec` unless we need to do something that's only supported in the `Gemfile`, so that they're detected when reading either source via the `gemspec` helper in the `Gemfile`.

Reply:
that seemed to be breaking [linting](https://github.com/gocardless/anony/actions/runs/6814441364/job/18531310939) for me, not quite sure why. happy to move if there's a way to disable the linting though.

Reply:
It would probably be worth making an org-wide decision on this and updating gc_ruboconfig, but you can change it on a per-project basis by adding the following to .rubocop.yml to make the linter complain if they're _not_ included in the gemspec with `add_development_dependency`
```
Gemspec/DevelopmentDependencies:
  EnforcedStyle: gemspec
```